### PR TITLE
Use multi-stage Docker build

### PR DESCRIPTION
**Pull request by praseodym**

### CODE REVIEW

File: docker/dockerbuild.sh [hunk 1]
```diff
@@ -1,25 +1,12 @@
...
 # Install programs required for build
-apk --no-cache add nodejs-lts git
 npm install -g yarn bower
 
 # Fetch dependencies
 yarn
 bower --allow-root install
 
 # Compress images and build
-yarn run build optimize-images
```
Comment (praseodym) [inline docker/dockerbuild.sh]: Is there a reason why `yarn build` was executed twice?

**Review by skamoen — APPROVED** [review]
Build log looks good, seems like a more realistic build process to me :)

File: docker/dockerbuild.sh [hunk 2]
```diff
@@ -12,14 +11,3 @@ bower --allow-root install
 # Compress images and build
 yarn run build optimize-images
 yarn run build
-cp -r ./build/. /srv/www
-
-# Uninstall installed build dependencies and remove cache
```
Comment (MatthijsKok) [inline docker/dockerbuild.sh:17]: Why are these no longer removed?

**Review by MatthijsKok — COMMENTED** [review]
How does this impact our build times? Running Docker in Travis might add unneeded overhead.

**Comment by skamoen** [comment]
No significant impact on build times

image

**Comment by MatthijsKok** [comment]
Ah wow. The Docker download files must be very well cached in Travis then.

File: docker/dockerbuild.sh (hunk 2 above)
Comment (praseodym) [reply docker/dockerbuild.sh:17]: Because the `builder` container is thrown away, only the artifacts that it built are used in the final image.

**Pull request merged**